src/linker/Linker.Dataflow/FlowAnnotations.cs

```diff
@@ -288,8 +288,9 @@ static bool IsTypeInterestingForDataflow (TypeReference typeReference)
 			// It's necessary to be able to implement a custom "Type" type in tests to validate
 			// the correct propagation of the annotations on "this" parameter. And tests can't really
 			// override System.Type - as it creates too many issues.
-			return (typeReference.Name.StartsWith ("Type") || typeReference.Name == "String") &&
-				typeReference.Namespace == "System";
+			// It also covers TypeBuilder and RuntimeType.
+			return typeReference.MetadataType == MetadataType.String ||
+				(typeReference.Name.Contains ("Type") && typeReference.Namespace.StartsWith ("System"));
```

Comment:
Is it ok that this includes types like https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.unmanagedtype?view=netcore-3.1 or https://docs.microsoft.com/en-us/dotnet/api/system.typedreference?view=netcore-3.1 ?

Reply:
We only do this filtering to help perf a bit to avoid looking for attributes on everything.

This just means that we're going to look for the dataflow annotations on locations typed like that. And if someone puts dataflow annotations on those places, it will be basically impossible to actually satisfy them. I don't have an opinion on this - it feels dirty to hardcode `RuntimeType` knowledge in the linker, but we can be more explicit here too...